gendiff/core.py

```diff
@@ -11,36 +11,57 @@ def get_keys_groups(
 
     union = keys1_set | keys2_set
     common = keys1_set & keys2_set
-    missing = keys1_set - common
 
     return {
         "union": union,
         "common": common,
-        "missing": missing,
+        "missing": keys1_set - common,
+        "new": keys2_set - common,
     }
 
 
-def get_diff_str(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> str:
+def get_diff_view(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> Dict[str, Any]:
     keys_groups = get_keys_groups(dict1, dict2)
-    output_str = "{\n"
+    diff_view = {}
+
+    for key in sorted(list(keys_groups["union"])):
+        if key in keys_groups["common"] and dict2[key] == dict1[key]:
+            diff_view[key] = "equal"
+        elif key in keys_groups["common"] and dict2[key] != dict1[key]:
+            if type(dict1[key]) == dict and type(dict2[key]) == dict:
+                diff_view[key] = get_diff_view(dict1[key], dict2[key])
+            else:
+                diff_view[key] = "common_diff"
+
+        elif key in keys_groups["missing"]:
+            diff_view[key] = "missing"
+        elif key in keys_groups["new"]:
+            diff_view[key] = "new"
+
+    return diff_view
 
-    for b in sorted(list(keys_groups["union"])):
-        if b in keys_groups["common"] and dict2[b] == dict1[b]:
-            output_str += f"      {b}: {dict2[b]}\n"
-        elif b in keys_groups["common"] and dict2[b] != dict1[b]:
-            output_str += f"    + {b}: {dict2[b]}\n"
-            output_str += f"    - {b}: {dict1[b]}\n"
-        elif b in keys_groups["missing"]:
-            output_str += f"    - {b}: {dict1[b]}\n"
-        else:
-            output_str += f"    + {b}: {dict2[b]}\n"
 
+def get_stylish_diff_format(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> str:
```

Comment:
I'd rename it to `get_diff_str` and make it to work on what `get_diff_view` returns (because you already have `generate_diff` that does all the job)